crates/ruff_linter/src/rules/ruff/rules/missing_await_for_coroutine.rs

```diff
@@ -0,0 +1,140 @@
+        format!("Coroutine is not awaited")
+    }
+
+    fn fix_title(&self) -> Option<String> {
+        Some(format!("Coroutine is not awaited"))
+    }
+}
+
+pub(crate) fn missing_await_for_coroutine(checker: &mut Checker, call: &ExprCall) {
+    // Only check for missing await in async context
+    if !checker.semantic().in_async_context() {
+        return;
+    }
+
+    // Try to detect possible scenarios where await is missing and ignore other cases
+    // For example, if the call is not a direct child of an statement expression or assignment statement
+    // then it's not reliable to determine if await is missing.
+    // User might return coroutine object from a function or pass it as an argument
+    if !possibly_missing_await(call, checker.semantic()) {
+        return;
+    }
+
+    let is_awaitable = is_awaitable_from_asyncio(call.func.as_ref(), checker.semantic())
+        || is_awaitable_func(call.func.as_ref(), checker.semantic());
+
+    // If call does not originate from asyncio or is not an async function, then it's not awaitable
+    if !is_awaitable {
+        return;
+    }
+
+    let mut diagnostic = Diagnostic::new(MissingAwaitForCoroutine {}, call.range);
+
+    diagnostic.set_fix(Fix::unsafe_edit(Edit::range_replacement(
+        checker.generator().expr(&generate_fix(call)),
+        call.range,
+    )));
+
+    checker.diagnostics.push(diagnostic);
+}
+
+fn is_awaitable_from_asyncio(func: &Expr, semantic: &SemanticModel) -> bool {
+    if let Some(call_path) = semantic.resolve_call_path(func) {
+        return matches!(
+            call_path.as_slice(),
+            ["asyncio", "sleep" | "wait" | "wait_for" | "gather"]
+        );
+    }
+    false
+}
+
+fn is_awaitable_func(func: &Expr, semantic: &SemanticModel) -> bool {
+    let Expr::Name(ExprName { id, .. }) = func else {
+        return false;
+    };
+    let Some(binding_id) = semantic.lookup_symbol(id) else {
+        return false;
+    };
+    let binding = semantic.binding(binding_id);
+    if let Some(node_id) = binding.source {
+        let node = semantic.statement(node_id);
+        if let Stmt::FunctionDef(StmtFunctionDef { is_async, name, .. }) = node {
+            return *is_async && name.as_str() == id;
+        }
+    }
+    false
+}
+
+fn possibly_missing_await(call: &ExprCall, semantic: &SemanticModel) -> bool {
+    if let Stmt::Expr(StmtExpr { value, .. }) = semantic.current_statement() {
+        if let Expr::Call(expr_call) = value.as_ref() {
+            return expr_call == call;
+        }
+    }
+
+    if let Some(Stmt::Assign(StmtAssign { value, .. })) = semantic.current_statement_parent() {
+        if let Expr::Call(expr_call) = value.as_ref() {
+            return expr_call == call;
+        }
+    }
+    false
+}
+
+/// Generate a [`Fix`] to add `await` for coroutine.
+///
+/// For example:
+/// - Given `asyncio.sleep(1)`, generate `await asyncio.sleep(1)`.
+fn generate_fix(call: &ExprCall) -> Expr {
```

Comment:
The function docs doesn't match what it actually returns. I think we should either return the `Fix` as mentioned in the docs or update the name to something like `create_await_expr`. I would recommend to go with the former.